### ISSUE TITLE
Integrate with Reciclanet's Ubuntu fsarchiver dump

### ISSUE THREAD

**Issue by txapelgorri** [eReuse/device-inventory#34]
Add an option to allow users to dump a preinstalled Ubuntu image, in order to obtain a plug-and-play computer.

**Comment by ivilata** [comment]
The script that @txapelgorri refers to is [this one](https://github.com/reciclanet/SCRIPTS/blob/master/instalar).

**Comment by ivilata** [comment]
I've performed several enhancements and fixes [here](https://github.com/eReuse/SCRIPTS/blob/ereuse/instalar), all suggested as PRs to the original. The new version is able to mount images from a CIFS/SMB server and use environment variables to override defaults in the script, to the point of performing an interactionless install. It also allows to dynamically compute the size of the root and swap partitions.

**Comment by txapelgorri** [comment]
I will tell Reciclanet to merge the PRs, but your plan with eReuse is to use the forked repository, right?

**Comment by ivilata** [comment]
@txapelgorri Yes, my plan is to contribute back all enhancements as PRs but in the meanwhile use our own ``ereuse`` integration branch so that no one's pressed to do the merges.

BTW, you'll surely face conflicts when merging the PRs one by one (many changes for such a short script). If you are ok with merging all the PRs, you may as well pull the changes from the ``ereuse`` branch and avoid some merging headaches I already took care of. `;)`

**Comment by txapelgorri** [comment]
Ok, I'll talk with Raimon this week and try merge all the PRs. Thanks for your effort! :)

**Comment by ivilata** [comment]
In its current state, the [reciclanet-install-34](https://github.com/eReuse/device-inventory/tree/reciclanet-install-34) branch fires, after the diagnostic/inventory and stress test, a shell script which reads configuration and invokes the ``instala`` script to automatically perform the installation.

It still needs testing on a prepared ISO and better integration to avoid the shell script (as is the case of the stress test).

**Comment by ivilata** [comment]
BTW, the new options added to ``config.ini`` are:

  - INSTALL=yes/no/ask
  - INSTALL_REMOTE_MP=(path of server to mount)
  - INSTALL_IMAGE_DIR=(directory inside of it containing the images)
  - INSTALL_IMAGE_NAME=(name of image to be automatically selected)

For instance, the following values select an image automatically for installation without questions:

```
INSTALL = yes
INSTALL_REMOTE_MP = /ereuse-data
INSTALL_IMAGE_DIR = images
INSTALL_IMAGE_NAME = lubuntu-i386-ca
```

**Comment by ivilata** [comment]
I just merged the branch into ``master`` to ease the integration of the script into Python.

**Comment by ivilata** [comment]
I merged the ``stress-install-integration`` branch into ``master``. Now the installation script is called straight from the Python script. I moved variables into the ``[installer]`` section of ``config.ini``, e.g.:

```
[installer]
INSTALL = yes
REMOTE_MP = /ereuse-data
IMAGE_DIR = images
IMAGE_NAME = lubuntu-i386-ca
```

The ``INSTALL`` and ``IMAGE_NAME`` options can be overridden via the ``--install`` and ``--image-name`` command line options.

**Comment by ivilata** [comment]
As a bonus, we could add a helper script (say ``di-save-image``) to dump the disk's root partition (if ext and not mounted) straight to the server's image directory.

**Comment by ivilata** [comment]
That script and the integration itself would be further simplified by #55.

**Comment by ivilata** [comment]
> As a bonus, we could add a helper script (say di-save-image) to dump the disk's root partition (if ext and not mounted) straight to the server's image directory.

Just created #63 for this particular task.
